core/views.py

```diff
@@ -251,6 +251,11 @@ def form_valid(self: "PostReportView", form: forms.ModelForm) -> HttpResponseRed
         messages.success(self.request, "Your post has been reported.")
         return super().form_valid(form)
 
+    def form_invalid(self: "PostReportView", form: forms.ModelForm) -> HttpResponse:
```

Comment:
Can you explain why this function is here? 
The default form invalid from django: https://github.com/django/django/blob/fd1dd767783b5a7ec1a594fcc5885e7e4178dd26/django/views/generic/edit.py#L65 is ok. Since we are in a browser based world there is no point in changing response code. Django is not changing response code for a reason so I would stay with that.
If your test expect 400 then tests are wrong and need fix, and since it is browser based world, they need to check response.context.form.errors.

Reply:
I don't understand the question. This function seems not needed at all.